src/main/java/net/minecraftforge/client/gui/widget/ForgeSlider.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Minecraft Forge - Forge Development LLC
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.client.gui.widget;
+
+import net.minecraft.client.gui.components.AbstractSliderButton;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.util.Mth;
+import org.lwjgl.glfw.GLFW;
+
+import java.text.DecimalFormat;
+
+/**
+ * Slider widget implementation which allows inputting values in a certain range with optional step size.
+ */
+public class ForgeSlider extends AbstractSliderButton
+{
+    protected Component prefix;
+    protected Component suffix;
+
+    protected double minValue;
+    protected double maxValue;
+
+    /** Allows input of discontinuous values with a certain step */
+    protected double stepSize;
+
+    protected boolean drawString;
+
+    private final DecimalFormat format;
+
+    /**
+     * @param precision Only used when {@code stepSize} is 0. Limited to a maximum of 4 (inclusive).
+     */
```

Comment:
Please document all params or none at all.

Reply:
My reasoning was that the rest of the parameters were pretty self explanatory, but sure, I can add the rest.

Reply:
Maybe to you, but the point of JDoc is not that the values don't need documentation if they are self-explanatory, but you know to document them.